psycop/common/global_utils/mlflow/mlflow_data_extraction.py

```diff
@@ -19,96 +25,91 @@ class MlflowAllMetricsFrame(ValidatedFrame[pl.DataFrame]):
+
+    @classmethod
+    def from_mlflow_run(
+        cls: type["PsycopMlflowRun"], run: Run, client: MlflowClient
+    ) -> "PsycopMlflowRun":
+        return cls(run_info=run._info, run_data=run._data, run_inputs=run._inputs, client=client)
+
+    def get_config(self) -> Config:
+        cfg_path = self.download_artifact(artifact_name="config.cfg", save_location=None)
+        return Config().from_disk(cfg_path)
+
+    def download_artifact(self, artifact_name: str, save_location: str | None = None) -> Path:
+        """Download an artifact from a run. Returns the path to the downloaded artifact.
+        If save_location is None, will save to temporary directory"""
+        return Path(
+            self._client.download_artifacts(
+                run_id=self.info.run_id, path=artifact_name, dst_path=save_location
+            )
+        )
 
-    allow_extra_columns = False
+    def get_all_metrics(self) -> pl.DataFrame:
+        return pl.DataFrame(self.data.metrics).with_columns(
+            pl.lit(self.info.run_name).alias("run_name")
+        )
 
 
-class MlflowMetricExtractor:
+class MlflowClientWrapper:
     def __init__(self) -> None:
-        self.client = MlflowClient(tracking_uri="http://exrhel0371.it.rm.dk:5050")
+        tracking_uri = "http://exrhel0371.it.rm.dk:5050"
+
+        self.client = MlflowClient(tracking_uri=tracking_uri)
+        mlflow.set_tracking_uri(tracking_uri)
 
     def get_all_metrics_for_experiment(self, experiment_name: str) -> MlflowAllMetricsFrame:
         """Get the final value of all logged metrics for each run in an experiment.
         Returns a long df with cols 'run_name', 'metric', and 'value'."""
         runs = self._get_mlflow_runs_by_experiment(experiment_name=experiment_name)
-        metrics_df = pl.concat([self._get_all_metrics_for_run(run) for run in runs]).melt(
+        metrics_df = pl.concat([run.get_all_metrics() for run in runs]).melt(
             id_vars="run_name", variable_name="metric"
         )
 
         return MlflowAllMetricsFrame(frame=metrics_df, allow_extra_columns=False)
 
-    def get_metrics_for_experiment(
-        self, experiment_name: str, metrics: Iterable[str]
-    ) -> MlflowTimeVaryingMetricsFrame:
-        """Get all values of specified metrics for each run in an experiment. This
-        includes values that get updated such as loss. Returns a long df with
-        cols 'run_name', 'metric', 'value', 'timestamp', 'step'."""
-        runs = self._get_mlflow_runs_by_experiment(experiment_name=experiment_name)
-        metrics_df = pl.concat(
-            [self._get_metrics_for_run(run=run, metrics=metrics) for run in runs]
-        ).rename({"key": "metric"})
-
-        return MlflowTimeVaryingMetricsFrame(frame=metrics_df, allow_extra_columns=False)
-
-    def get_best_run_from_experiments(self, experiment_names: Iterable[str], metric: str) -> Run:
+    def get_best_run_from_experiment(self, experiment_name: str, metric: str) -> PsycopMlflowRun:
```

Comment:
Vi antager at "godt" er "større" for metric'en. Skal vi måske have en direction på?

Reply:
Good point!